### PR TITLE
Apply Remove overridden assignment JDT clean in core.commands

### PR DESCRIPTION
**Pull request by vogella**
Dogfooding the improve JDT cleanup from Bug 572440 to ensure the JDT
cleanup is stable and functional. Push in small chunks to make review
easier.

### CODE REVIEW

File: bundles/org.eclipse.core.commands/src/org/eclipse/core/commands/ParameterizedCommand.java [hunk 1]
```diff
@@ -307,10 +307,7 @@ public static ParameterizedCommand generateCommand(Command command, Map paramete
 			ArrayList<Parameterization> parms = new ArrayList<>();
 			for (Entry<?, ?> entry : ((Map<?, ?>) parameters).entrySet()) {
 				String key = (String) entry.getKey();
-				IParameter parameter = null;
-				// get the parameter from the command
```
Comment (vogella) [inline bundles/org.eclipse.core.commands/src/org/eclipse/core/commands/ParameterizedCommand.java:311]: comment lost, will manually update the change

**Review by vogella — COMMENTED** [review]
See comment

**Comment by mickaelistria** [comment]
Requires a version bump

**Comment by vogella** [comment]
> Requires a version bump

Done via another commit / PR

**Pull request merged**